matsim/src/main/java/org/matsim/core/population/routes/PopulationComparison.java

```diff
@@ -3,127 +3,185 @@
-		if (o1 instanceof Leg) {
-			if (o2 instanceof Leg) {
-				Leg leg1 = (Leg) o1;
-				Leg leg2 = (Leg) o2;
-				if (!leg1.getDepartureTime().equals(leg2.getDepartureTime())) {
-					return false;
-				}
-				if (!leg1.getMode().equals(leg2.getMode())) {
-					return false;
-				}
-				if (!leg1.getTravelTime().equals(leg2.getTravelTime())) {
-					return false;
-				}
-			} else {
-				return false;
-			}
-		} else if (o1 instanceof Activity) {
-			if (o2 instanceof Activity) {
-				Activity activity1 = (Activity) o1;
-				Activity activity2 = (Activity) o2;
-				if (activity1.getEndTime().isUndefined() ^ activity2.getEndTime().isUndefined()) {
-					return false;
-				}
-				if (activity1.getEndTime().isDefined() && activity1.getEndTime().seconds()
-					!= activity2.getEndTime().seconds()) {
-					return false;
-				}
-				if (activity1.getStartTime().isUndefined() ^ activity2.getStartTime().isUndefined()) {
-					return false;
-				}
-				if (activity1.getStartTime().isDefined() && activity1.getStartTime().seconds()
-					!= activity2.getStartTime().seconds()) {
-					return false;
-				}
-			} else {
-				return false;
-			}
-		} else {
-			throw new RuntimeException ("Unexpected PlanElement");
-		}
-		return true;
-	}
+public class PopulationComparison {
+
+    private final static double DEFAULT_DELTA = 1e-10;
+
+    public enum Result {equal, notEqual}
+
+    private static final Logger log = LogManager.getLogger(PopulationComparison.class);
+
+    private PopulationComparison() {
+    }
+
+    public static Result compare(Population population1, Population population2) {
+        return compare(population1, population2, DEFAULT_DELTA);
+    }
+
+
+    public static Result compare(Population population1, Population population2, double delta) {
+        Result result = Result.equal;
+
+        Iterator<? extends Person> it1 = population1.getPersons().values().iterator();
+        Iterator<? extends Person> it2 = population2.getPersons().values().iterator();
+
+        while (it1.hasNext() || it2.hasNext()) {
+            if (!it1.hasNext()) {
+                result = Result.notEqual;
+                log.warn("");
+                log.warn(" different length in populations. ");
+                return result;
+            }
+            if (!it2.hasNext()) {
+                result = Result.notEqual;
+                log.warn("");
+                log.warn(" different length in populations. ");
+                return result;
+            }
+
+            Person person1 = it1.next();
+            Person person2 = it2.next();
+
+            if (!person1.getId().equals(person2.getId())) {
+                log.warn("");
+                log.warn("persons out of sequence p1: " + person1.getId() + " | p2: " + person2.getId());
+                result = Result.notEqual;
+                continue;
+            }
+
+            if (!AttributesComparison.equals(person1.getAttributes(), person2.getAttributes())) {
+                log.warn("");
```

Comment:
empty output, is it a way of injecting newlines into logs for readability?

Reply:
yes I copied that from the previous behavior of this class